### ISSUE TITLE
no feature 'bareword_filehandles' mistakes package name for bareword filehandle

### ISSUE THREAD

**Issue by oschwald** [Perl/perl5#19957]
This code:

```
use v5.36;
no feature 'bareword_filehandles';
use Data::Dumper;

say Data::Dumper->Dump( ['test'] );
```

produces this compilation error:

```
Bareword filehandle "Data::Dumper" not allowed under 'no feature "bareword_filehandles"' at t.pl line 5.
Execution of t.pl aborted due to compilation errors.
```

I noticed this in 5.34, but I thought https://github.com/Perl/perl5/pull/19272 would fix it.

**Perl configuration**

```
Summary of my perl5 (revision 5 version 36 subversion 0) configuration:

  Platform:
    osname=linux
    osvers=5.15.0-40-generic
    archname=x86_64-linux
    uname='linux greg-thinkpad-t480 5.15.0-40-generic #43-ubuntu smp wed jun 15 12:54:21 utc 2022 x86_64 x86_64 x86_64 gnulinux '
    config_args='-Dprefix=/home/greg/.plenv/versions/5.36.0 -de -Dversiononly -A'eval:scriptdir=/home/greg/.plenv/versions/5.36.0/bin''
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=undef
    usemultiplicity=undef
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
  Compiler:
    cc='cc'
    ccflags ='-fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    optimize='-O2'
    cppflags='-fwrapv -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion=''
    gccversion='11.2.0'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='cc'
    ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib/x86_64-linux-gnu /usr/lib /usr/lib64
    libs=-lpthread -lnsl -ldb -ldl -lm -lcrypt -lutil -lc
    perllibs=-lpthread -lnsl -ldl -lm -lcrypt -lutil -lc
    libc=/lib/x86_64-linux-gnu/libc.so.6
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version='2.35'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E'
    cccdlflags='-fPIC'
    lddlflags='-shared -O2 -L/usr/local/lib -fstack-protector-strong'


Characteristics of this binary (from libperl):
  Compile-time options:
    HAS_TIMES
    PERLIO_LAYERS
    PERL_COPY_ON_WRITE
    PERL_DONT_CREATE_GVSV
    PERL_MALLOC_WRAP
    PERL_OP_PARENT
    PERL_PRESERVE_IVUV
    USE_64_BIT_ALL
    USE_64_BIT_INT
    USE_LARGE_FILES
    USE_LOCALE
    USE_LOCALE_COLLATE
    USE_LOCALE_CTYPE
    USE_LOCALE_NUMERIC
    USE_LOCALE_TIME
    USE_PERLIO
    USE_PERL_ATOF
  Built under linux
  Compiled at Jul 13 2022 14:41:08
  %ENV:
    PERL5LIB=":/home/greg/MaxMind/mm_website/lib:/home/greg/MaxMind/mm_website/t/lib:/home/greg/MaxMind/mm_website/lab/lib"
  @INC:
    /home/greg/MaxMind/mm_website/lib
    /home/greg/MaxMind/mm_website/t/lib
    /home/greg/MaxMind/mm_website/lab/lib
    /home/greg/.plenv/versions/5.36.0/lib/perl5/site_perl/5.36.0/x86_64-linux
    /home/greg/.plenv/versions/5.36.0/lib/perl5/site_perl/5.36.0
    /home/greg/.plenv/versions/5.36.0/lib/perl5/5.36.0/x86_64-linux
    /home/greg/.plenv/versions/5.36.0/lib/perl5/5.36.0
```

**Comment by xenu** [comment]
This a duplicate of #19704, which was fixed in blead with commit c5327cb77526e4e4c7bb83e24be20ac48e85b1fa.

That bug is the reason why we made `use v5.36` not disable `bareword_filehandles`.

**Comment by oschwald** [comment]
Thanks! I apologize for missing that in my search.
